### ISSUE TITLE
Syntax highlighting after pattern with start condition and escaped quote incorrect

### ISSUE THREAD

**Issue by rveerd** [babyraging/yash#1]
In the following Lex input file syntax highlighting stops working correctly after the pattern in the rule returning "STRING":

```lex
NUMBER      [0-9]+
ID          [$A-Za-z_][$0-9A-Za-z_]*
DOTTED_ID   {ID}("."{ID})*

%x TAG

%%

<TAG>{NUMBER}           { return "NUMBER"; }
<TAG>\"(\\\"|[^"])*\"   { return "STRING"; } /* error */
<TAG>{DOTTED_ID}        { return "ID"; }
<TAG>"@"{DOTTED_ID}     { return "PARAM"; }
```

If I remove the `<TAG>` or trailing `\"` in the pattern, syntax highlighting works correctly.

Yash Version: 0.2.0
VS Code Version: 1.44.2 (user setup)
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T16:36:23.138Z (1 wk ago)
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by babyraging** [comment]
Hi rveerd, thank you for the issue.

Indeed this is wrong, I can reproduce it, thanks again for finding it.

Maybe tonight or tomorrow I'll have the issue fixed.

**Comment by babyraging** [comment]
The issue should be fixed with 0.2.1.
